esphome/components/voice_kit/voice_kit.h

```diff
@@ -29,11 +90,47 @@ class VoiceKit : public Component, public i2c::I2CDevice {
 
   void set_reset_pin(GPIOPin *reset_pin) { reset_pin_ = reset_pin; }
 
-  bool dfu_get_version_();
+  void set_firmware_bin(const uint8_t *data, const uint32_t len) {
+    this->firmware_bin_ = data;
+    this->firmware_bin_length_ = len;
+  }
+  void set_firmware_version(const uint8_t major, const uint8_t minor, const uint8_t patch) {
```

Comment:
If clang-tidy were here, it would be very disappointed with you :rofl: 
```suggestion
  void set_firmware_version(uint8_t major, uint8_t minor, uint8_t patch) {
```